src/Management/management/Api/IndicesApi.php

```diff
@@ -0,0 +1,2933 @@
+                    $e->getResponse() ? $e->getResponse()->getBody()->getContents() : null
+                );
+            }
+
+            $statusCode = $response->getStatusCode();
+
+            if ($statusCode < 200 || $statusCode > 299) {
+                throw new ApiException(
+                    sprintf(
+                        '[%d] Error connecting to the API (%s)',
+                        $statusCode,
+                        $request->getUri()
+                    ),
+                    $statusCode,
+                    $response->getHeaders(),
+                    $response->getBody()
+                );
+            }
+
+            $responseBody = $response->getBody();
+            if ($returnType === '\SplFileObject') {
+                $content = $responseBody; //stream goes to serializer
+            } else {
+                $content = $responseBody->getContents();
+                if (!in_array($returnType, ['string','integer','bool'])) {
+                    $content = json_decode($content);
+                }
+            }
+
+            return [
+                ObjectSerializer::deserialize($content, $returnType, []),
+                $response->getStatusCode(),
+                $response->getHeaders()
+            ];
+
+        } catch (ApiException $e) {
+            switch ($e->getCode()) {
+                case 200:
+                    $data = ObjectSerializer::deserialize(
+                        $e->getResponseBody(),
+                        '\Swagger\Client\Model\ReindexingTask',
+                        $e->getResponseHeaders()
+                    );
+                    $e->setResponseObject($data);
+                    break;
+            }
+            throw $e;
+        }
+    }
+
+    /**
+     * Operation getReindexingStatusAsync
+     *
+     * Return the status of the current reindexing task.
+     *
+     * @param  string $hashid Unique id of a search engine. (required)
+     * @param  string $name Name of an index. (required)
+     *
+     * @throws \InvalidArgumentException
+     * @return \GuzzleHttp\Promise\PromiseInterface
+     */
+    public function getReindexingStatusAsync($hashid, $name)
+    {
+        return $this->getReindexingStatusAsyncWithHttpInfo($hashid, $name)
+            ->then(
+                function ($response) {
+                    return $response[0];
+                }
+            );
+    }
+
+    /**
+     * Operation getReindexingStatusAsyncWithHttpInfo
+     *
+     * Return the status of the current reindexing task.
+     *
+     * @param  string $hashid Unique id of a search engine. (required)
+     * @param  string $name Name of an index. (required)
+     *
+     * @throws \InvalidArgumentException
+     * @return \GuzzleHttp\Promise\PromiseInterface
+     */
+    public function getReindexingStatusAsyncWithHttpInfo($hashid, $name)
+    {
+        $returnType = '\Swagger\Client\Model\ReindexingTask';
```

Comment:
if you expect class name, it's better to use class constant like \Swagger\Client\Model\ReindexingTask::class